docs(apig): update fields description (#3716)
huaweicloud/terraform-provider-huaweicloud: docs/resources/apig_api.md

  The valid values are **Public** and **Private**.

* `name` - (Required, String) Specifies the API name.  
  The valid length is limited from can contain `3` to `255`, only Chinese and English letters, digits,  hyphens (-),  
  underscores(_),  periods (.),  slashes,  commas in Chinese formats, parentheses and colons in Chinese and English formats

>   are allowed.  
>   The name must start with digit, Chinese or English letter.

Reviewer: ```suggestion
  The name must start with a digit, Chinese or English letter.
```

Author: fixed
